src/screens/HyperSwitch/Connectors/ConnectorAccountDetailsHelper.res

```diff
@@ -56,6 +58,63 @@ let inputField = (
     (),
   )
 
+let inputFieldForConnectorLabel = (
+  ~name,
+  ~field,
+  ~label,
+  ~connector: ConnectorTypes.connectorName,
+  ~getPlaceholder,
+  ~checkRequiredFields,
+  ~disabled,
+) => {
+  open MerchantAccountUtils
+  let businessProfiles = HyperswitchAtom.businessProfilesAtom->Recoil.useRecoilValueFromAtom
+  let defaultBusinessProfile = businessProfiles->getValueFromBusinessProfile
+  let arrayOfBusinessProfile = businessProfiles->getArrayOfBusinessProfile
+  let connectorName = connector->ConnectorUtils.getConnectorNameString
+  let profileId =
+    ReactFinalForm.useField(`profile_id`).input.value->LogicUtils.getStringFromJson("")
+  let profileName = (
+    arrayOfBusinessProfile
+    ->Js.Array2.find((ele: HSwitchSettingTypes.profileEntity) => ele.profile_id === profileId)
+    ->Belt.Option.getWithDefault(defaultBusinessProfile)
+  ).profile_name
+  let connectorLabelOnChange = ReactFinalForm.useField(`connector_label`).input.onChange
+
+  React.useEffect1(() => {
```

Comment:
Can't we set this default value on the time of initial generation itself

Reply:
We checked with that flow but it is not re-rendering as the flow we want and if we apply memos in that it was making the profile dropdown issue as they all are using same component.

Reply:
Updated